### ISSUE TITLE
[DOC] Architecture Diagram

### ISSUE THREAD

**Issue by m-ildefons** [longhorn/longhorn#6761]
## What's the document you plan to update? Why? Please describe

Create an architecture diagram showing all Longhorn components their interactions and the separate data- and control-planes.

## Additional context

As an example, a similar diagram for OpenStack:
![image](https://github.com/longhorn/longhorn/assets/17141774/5e8d0997-4394-4ec3-a587-ead34686f6d7)

**Comment by derekbit** [comment]
Some thoughts. We can have macroscopic and microscopic architecture diagrams. 

- Macroscopic architecture diagram: This type of diagram provides a high-level overview of the system. It typically includes components such as pods, ports, labels, and the overall connectivity between them. This diagram is useful for understanding the structure and topology of the system.

- Microscopic architecture diagram: This diagram focuses on a more detailed level, highlighting the interactions between individual components within the system. It could show how specific parts communicate, data flow, and dependencies.

**Comment by shuo-wu** [comment]
Can the macroscopic diagram be what we use in the speaking sessions? 
<img width="1070" alt="Screenshot 2024-04-12 at 21 21 04" src="https://github.com/longhorn/longhorn/assets/47542419/dc76f5f3-37be-433a-9ed7-2e256cf8a7f8">

**Comment by derekbit** [comment]
> Can the macroscopic diagram be what we use in the speaking sessions? <img alt="Screenshot 2024-04-12 at 21 21 04" width="1070" src="https://private-user-images.githubusercontent.com/47542419/322176799-dc76f5f3-37be-433a-9ed7-2e256cf8a7f8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI5ODMzOTAsIm5iZiI6MTcxMjk4MzA5MCwicGF0aCI6Ii80NzU0MjQxOS8zMjIxNzY3OTktZGM3NmY1ZjMtMzdiZS00MzNhLTllZDctMmUyNTZjZjhhN2Y4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDEzVDA0MzgxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZiMDMzMTcwYmY1MzY3NDQ4MjYwOTlhZWYyNzg2ODNkMWIzZTc0NGQ1OWE1N2Q2ZTRjNDE1MjZmY2RmYzQzOTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.C11XvBTXCdOvAl8bYSMxOIbZltO-resC7JF2TKTVCQA">

Yes! Looks good. And we can circle replicas and engine to form an instance-manager pod.
I'm thinking we can add some information such as Longhorn's labels and ports in the macroscopic diagram as well.

**Comment by chriscchien** [comment]
Verified **pass** on longhorn website master(https://github.com/longhorn/website/commit/edae746c6805f430437b8a2c8fbc06a89c65a4b2)

Architecture and Concepts content updated in below branches and https://longhorn.io/img/diagrams/architecture/how-longhorn-works-with-kubernetes.svg displayed correctly
- https://longhorn.io/docs/1.6.0/concepts/
- https://longhorn.io/docs/1.6.1/concepts/
- https://longhorn.io/docs/1.6.2/concepts/
- https://longhorn.io/docs/1.6.3/concepts/
- https://longhorn.io/docs/1.6.4/concepts/
- https://longhorn.io/docs/1.7.0/concepts/
- https://longhorn.io/docs/1.7.1/concepts/
- https://longhorn.io/docs/1.7.2/concepts/
- https://longhorn.io/docs/1.7.3/concepts/
- https://longhorn.io/docs/1.8.0/concepts/
- https://longhorn.io/docs/1.8.1/concepts/
